### PR TITLE
refactor: implement `get_header_as_int` method

### PR DESCRIPTION
**Pull request by vgerak**
# Summary of Changes

Implements the `get_header_as_int` method.

Wasn't sure whether (and how) to document it in the docs, as I couldn't find the part for the similar function `get_header_as_datetime()`.

# Related Issues

#2060

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [x] Added **tests** for changed code.
- [x] Prefixed code comments with GitHub nick and an appropriate prefix.
- [x] Coding style is consistent with the rest of the framework.
- [x] Updated **documentation** for changed code.
    - [x] Added docstrings for any new classes, functions, or modules.
    - [x] Updated docstrings for any modifications to existing code.
    - [x] Updated both WSGI and ASGI docs (where applicable).
    - [ ] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [ ] Updated all relevant supporting documentation files under `docs/`.
    - [x] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [x] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/latest/quickstart.html#creating-news-fragments) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

If you have *any* questions to *any* of the points above, just **submit and ask**! This checklist is here to *help* you, not to deter you from contributing!

*PR template inspired by the attrs project.*

### CODE REVIEW

**Comment by vgerak** [comment]
@vytas7 pinging issue creator

**Comment by vytas7** [comment]
Hi @vgerak!
And thanks for this PR, and sorry I've been swamped with other things recently not directly related to Falcon in any way, so I couldn't review in a timely fashion. (If you want to count this towards Hacktoberfest, I've tentatively added the relevant label that should do the job.)

It seems that there are unrelated issues with our CI which must be addressed prior to proceeding with your PR, I'll try to prioritize this when I find some free time.

File: tests/test_headers.py [hunk 1]
```diff
@@ -363,6 +363,25 @@ def test_expires_header(self, client):
 
         assert result.headers['Expires'] == 'Tue, 01 Jan 2013 10:30:30 GMT'
 
+    def test_get_header_as_int(self, client):
+        resource = testing.SimpleTestResource(body=SAMPLE_BODY)
+        client.app.add_route('/', resource)
+        request_headers = {
+            'X-Int-Val': '42',
+            'X-Str-Val': 'test-val',
```
Comment (kgriffs) [inline tests/test_headers.py:371]: Nitpick: For good measure also test that it fails with a fractional number, e.g. `'3.14'`

File: tests/test_headers.py [hunk 2]
```diff
@@ -363,6 +363,25 @@ def test_expires_header(self, client):
...
+        }
+        client.simulate_get(headers=request_headers)
+
+        req = resource.captured_req
+        value = req.get_header_as_int('X-Int-Val')
+        assert value == 42
+
+        value = req.get_header_as_int('X-Not-Found')
```
Comment (kgriffs) [inline tests/test_headers.py:379]: Also test with `required=True`

**Comment by vgerak** [comment]
rebased and pushed the changes to the branch, but for some reason github isn't picking up the latest version of the branch..

**Review by vytas7 — APPROVED** [review]
LGTM :+1:

**Review by CaselIT — APPROVED** [review]

**Pull request merged**